tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/modules/module_table_name_test.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:serverpod_cli/src/analyzer/models/stateful_analyzer.dart';
+import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
+import 'package:serverpod_cli/src/test_util/builders/model_source_builder.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test(
+      'Given a module class with the same table name as a user defined table then there is an error reported.',
+      () {
+    var models = [
+      ModelSourceBuilder()
+          .withModuleAlias('auth')
+          .withFileName('user_info')
+          .withYaml(
+        '''
+        class: UserInfo
+        table: serverpod_user_info
+        fields:
+          nickname: String
+        ''',
+      ).build(),
+      ModelSourceBuilder()
+          .withModuleAlias('protocol')
+          .withFileName('user_info')
+          .withYaml(
+        '''
+        class: User
+        table: serverpod_user_info
+        fields:
+          name: String
+        ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    StatefulAnalyzer analyzer = StatefulAnalyzer(
+      models,
+      onErrorsCollector(collector),
+    );
+    analyzer.validateAll();
+    var errors = collector.errors;
+
+    expect(errors, isNotEmpty);
+
+    expect(
+      errors.first.message,
+      contains(
+        'The table name "serverpod_user_info" is already in use by the class "User".',
```

Comment:
Question: Is the priority always the project defined tables before module defined tables?

In that case, maybe we should give some more hints in the error message if the collision table is part of a module package.
Since the user does not controll the name of the module tables it might also be better to switch this priority so that we hint the user to change the name of the project defined table instaed.

On another note, this is a pretty sucky scenario to run into if you add a module as a dependency in a later stage project.

Reply:
Not sure I get what you mean by priority?

It works like this:

Each file is analyzed in isolation and will show any relevant errors, order doesn't matter. 

This error will be reported in a model that has defined `table: serverpod_user_info` in your project. This means the user gets this info in their file.

Reply:
Aaah gotcha! User will only see this message in their own files. Then would it be usefully to give a hint towards which module the table belongs to? Might be hard to understand where the class "User" comes from or what could have caused the collission?

On that note, maybe modules should always have their tables prefixed or separated in some way to avoid these type of issues.

Reply:
They have their tables prefixed actually, this is why they all start with `serverpod_` this prefix is however manual but still!